Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Coloring
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Concrete colorings of common graphs
+
+This file defines colorings for some common graphs
+
+## Main declarations
+
+* `SimpleGraph.pathGraph.bicoloring`: Bicoloring of a path graph
+-/
+
+theorem suc_mod2_neq (m : ℕ) : (m % 2 == 0) ≠ ((m + 1) % 2 == 0) := by
+  intro h
+  rw [Bool.eq_iff_eq_true_iff, beq_iff_eq] at h
+  simp only [beq_iff_eq, ←Nat.even_iff, Nat.even_add_one] at h
+  exact not_iff_self h.symm
+
+/-- Bicoloring of a path graph -/
+def SimpleGraph.pathGraph.bicoloring (n : ℕ) :
+    SimpleGraph.Coloring (pathGraph n) Bool :=
+  Coloring.mk (fun u ↦ u.val % 2 == 0)
+    (by
+      intro u v h
+      simp
+      have : u ⋖ v ∨ v ⋖ u := by simp_all [pathGraph]
+      have : u.val ⋖ v.val ∨ v.val ⋖ u.val := by simp_all [Fin.coe_covby_iff]
+      have : u.val + 1 = v.val ∨ v.val + 1 = u.val := by simp_all [Nat.covby_iff_succ_eq]
+      match this with
+        | Or.inl h' =>
+          rw [←h']
+          exact suc_mod2_neq u
+        | Or.inr h' =>
+          rw [←h']
+          exact (suc_mod2_neq v).symm)
+
+/-- Convert a coloring to bool to a coloring to Fin 2 -/
+def SimpleGraph.Coloring.BoolToFin2 {α} {G : SimpleGraph α} (c : SimpleGraph.Coloring G Bool) :
+    SimpleGraph.Coloring G (Fin 2) :=
+  (SimpleGraph.recolorOfEquiv G (finTwoEquiv)).invFun c
+
+theorem SimpleGraph.pathGraph.clique (n : ℕ) (h : n > 1) :
+    SimpleGraph.IsClique (pathGraph n) {⟨0, Nat.zero_lt_of_lt h⟩, ⟨1, h⟩} := by
+  let s : Finset (Fin n) := {⟨0, Nat.zero_lt_of_lt h⟩, ⟨1, h⟩}
+  have hs : SimpleGraph.IsClique (pathGraph n) s := by
+    refine (pairwise_subtype_iff_pairwise_set s (pathGraph n).Adj).mp ?_
+    intro (x : s) (y : s) (hxy : x ≠ y)
+    -- Adj (pathGraph n) ↑x ↑y
+    simp [pathGraph]
+    -- ↑x ⋖ ↑y ∨ ↑y ⋖ ↑x
+    have : (x : ℕ) = 0 ↔ (y : ℕ) = 1 := by aesop
+    have : (x : ℕ) = 1 ↔ (y : ℕ) = 0 := by aesop
+    have x_val : (x : ℕ) = 0 ∨ (x : ℕ) = 1 := by aesop
+    apply Or.elim x_val
+    repeat
+      intro _
+      simp_all [Fin.coe_covby_iff.symm, Nat.covby_iff_succ_eq.mpr rfl]
+  simp_all [Finset.mem_singleton, Fin.mk.injEq, Finset.coe_insert,
+            Finset.coe_singleton, Set.mem_singleton_iff]
```

Comment:
Rather than using cliques, I find working with graph embeddings to be preferable (maps are more fun than sets, right?) Here's the main content of this clique lemma, but it lets you avoid needing to construct this two-element finset, and `⊤` (the complete graph) is what a clique is modeling.
```lean
def pathGraph_two_embedding (n : ℕ) (h : 2 ≤ n) : pathGraph 2 ↪g pathGraph n where
  toFun v := ⟨v, trans v.2 h⟩
  inj' := by
    rintro v w
    rw [Fin.mk.injEq]
    exact Fin.ext
  map_rel_iff' := by
    intro v w
    fin_cases v <;> fin_cases w <;> simp [pathGraph, ← Fin.coe_covby_iff]

-- move to right after `pathGraph`
theorem pathGraph_two_eq_top : pathGraph 2 = ⊤ := by
  ext u v
  fin_cases u <;> fin_cases v <;> simp [pathGraph, ← Fin.coe_covby_iff, Nat.covby_iff_succ_eq]
```
Style comments: Note the non-terminal simp. Also note that `simp_all` obscures that `hs` is essential to closing the goal. In this case I believe you can use `simpa using hs`, which is a way to do `simp` on `hs` and the goal simultaneously, and then it does `exact` for you with the simplified `hs`.